components/restapi/core/blocks.go

```diff
@@ -33,6 +33,20 @@ func blockMetadataByBlockID(blockID iotago.BlockID) (*api.BlockMetadataResponse,
 	return blockMetadata.BlockMetadataResponse(), nil
 }
 
+func transactionMetadataByBlockID(blockID iotago.BlockID) (*api.TransactionMetadataResponse, error) {
+	blockMetadata, err := deps.Protocol.MainEngineInstance().Retainer.BlockMetadata(blockID)
+	if err != nil {
+		return nil, ierrors.Wrapf(echo.ErrInternalServerError, "failed to get block metadata %s: %s", blockID.ToHex(), err)
+	}
+
+	metadata := blockMetadata.TransactionMetadataResponse()
+	if metadata == nil {
+		return nil, ierrors.Wrapf(echo.ErrNotFound, "transaction not found")
```

Comment:
Please add the blockid here.
```suggestion
		return nil, ierrors.Wrapf(echo.ErrNotFound, "no transaction found for block ID %s", blockID.ToHex())
```